Sources/DeepLearning/Layer.swift

```diff
@@ -1295,3 +1295,59 @@ public struct Reshape<Scalar: TensorFlowFloatingPoint>: Layer {
         return input.reshaped(toShape: shape)
     }
 }
+
+struct LSTMCell<Scalar: TensorFlowFloatingPoint>: Layer {
+
+    var inputWeight: Tensor<Scalar>
+    var updateWeight: Tensor<Scalar>
+    var forgetWeight: Tensor<Scalar>
+    var forgetBias: Tensor<Scalar>
+    var outputWeight: Tensor<Scalar>
+
+    init(inputSize: Int32, hiddenSize: Int32) {
+        let concatenatedInputSize = inputSize + hiddenSize
+        self.inputWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+        self.updateWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+        self.forgetWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+        self.forgetBias = Tensor<Scalar>(zeros: [hiddenSize])
+        self.outputWeight = Tensor<Scalar>(glorotUniform: [concatenatedInputSize, hiddenSize])
+    }
+
+    struct HiddenState: Differentiable {
+        var cellState: Tensor<Scalar>
+        var hiddenState: Tensor<Scalar>
+
+        @differentiable
+        init(cellState: Tensor<Scalar>, hiddenState: Tensor<Scalar>) {
+            self.cellState = cellState
+            self.hiddenState = hiddenState
+        }
+    }
+
+    struct Input: Differentiable {
+        var inputs: Tensor<Scalar>
```

Comment:
We should find a more meaningful name than “inputs” which confusingly repeats the name of the parent type “Input”.

Reply:
I'm not too sure, or I'm not creative enough to come up with a name for it. The input to the LSTM layer consists of:

(a) the input for _that_ timestep.
(b) the previous timestep's hidden state.

What do you think we could name it?

Reply:
I don't have a good answer here, but anything that differentiates the time step input from `Input` would be less confusing. How about the following?
```swift
public struct Input: Differentiable {
    public stepInput: Tensor<Scalar>
    public state: State
}
```

I don't feel strongly at all about `stepInput`. Feel free to suggest a better name.

Reply:
how about ```currentstate``` or ```currentInput``` or we could go change convention and go to something like ```c_0``` and ```h_0```, denoting current input and hidden state input